homeassistant/components/systemmonitor/sensor.py

```diff
@@ -166,6 +183,20 @@ def update(self):
             self._state = round(psutil.swap_memory().free / 1024 ** 2, 1)
         elif self.type == "processor_use":
             self._state = round(psutil.cpu_percent(interval=None))
+        elif self.type == "processor_temperature":
+            temps = psutil.sensors_temperatures()
+            # There might be additional keys to be added for different
+            # platforms / hardware combinations.
+            # Raspberry 4
+            if "cpu-thermal" in temps:
+                self._state = round(temps["cpu-thermal"][0].current, 1)
+            # Linux default
+            elif "coretemp" in temps:
+                self._state = round(temps["coretemp"][0].current, 1)
+            # Fallback if no CPU sensor data available
+            else:
+                self._state = "N/A"
+                self._unit_of_measurement = None
```

Comment:
I did that since otherwise it would output "N/A °C". I wasn't aware that the unit is a "key" for he history and that it is therefore unsafe to change.

Reply:
The state should not be `N/A`, if the state is not available or unknown, `None` should be returned.

 Availability can be implemented separately using the `available` properties of an entity.

Reply:
Okay, I will change the `state` and also look into `available`. Thanks for pointing that out!